platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/AbstractProfileCommand.java

```diff
@@ -0,0 +1,229 @@
+        this.applicationService = applicationService;
+    }
+    protected void setFeaturesService(FeaturesService featuresService) {
+        this.featuresService = featuresService;
+    }
+    protected void setBundleService(BundleService bundleService) {
+        this.bundleService = bundleService;
+    }
+
+
+    @Override
+    public Object execute() throws Exception {
+
+        if (applicationService == null) {
+            throw new IllegalStateException("ApplicationService not found");
+        }
+        if (featuresService == null) {
+            throw new IllegalStateException("FeaturesService not found");
+        }
+        if (bundleService == null) {
+            throw new IllegalStateException("BundleService not found");
+        }
+        try {
+            doExecute(applicationService, featuresService, bundleService);
+        } catch (ApplicationServiceException ase) {
+            printError("Encountered error while trying to perform command. Check log for more details.");
+            LOGGER.debug("Error while performing command.", ase);
+        }
+        return null;
+    }
+
+    /**
+     * Execute profile interaction
+     * be sent to the console and nothing should be returned.
+     *
+     * @param applicationService
+     * @param featuresService
+     * @param bundleService
+     */
+    protected abstract void doExecute(ApplicationService applicationService,
+                            FeaturesService featuresService,
+                            BundleService bundleService)
+            throws Exception;
+
+
+    /**
+     * Get the {@link Feature} that makes up a profile
+     * @param applicationService {@link ApplicationService}
+     * @param profileName Name of profile to get
+     * @return {@link Feature}
+     */
+    protected Optional<Feature> getProfile(ApplicationService applicationService, String profileName) {
+        return Optional.ofNullable(applicationService.getInstallationProfiles().stream()
+                .filter(i -> i.getName().replaceFirst("profile-", "").equals(profileName))
+                .findFirst().orElse(null));
+    }
+
+    /**
+     * Gets the content of an advanced profile
+     * @param profileName Name of profile to get
+     * @return {@link Map}
+     */
+    protected Optional<Map<String, List<String>>> getProfile(String profileName) {
+        File profileFile = Paths.get(profilePath.toString(), profileName + ".json").toFile();
+        Map<String, List<String>> profileMap = null;
+
+        try {
+            profileMap = (Map<String, List<String>>) fromJson(FileUtils.readFileToString(profileFile, StandardCharsets.UTF_8));
+        } catch (FileNotFoundException e) {
+            LOGGER.debug("The file associated with profile: {} was not found under {}",
+                    profileName, profilePath);
+        } catch (IOException e) {
+            LOGGER.debug("An IOException occurred: ", e);
```

Comment:
❓ Should we ignore here or let the exception bubble up? Looks like the message `Profile: %s not found` if this exception occurs right now which might be misleading. Thoughts?